subsys/bluetooth/services/fast_pair/fmdn/motion_detector.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(fp_fmdn_motion_detector, CONFIG_BT_FAST_PAIR_LOG_LEVEL);
+
+#include <bluetooth/services/fast_pair/fast_pair.h>
+#include <bluetooth/services/fast_pair/fmdn.h>
+#include <dult.h>
+
+#include "fp_fmdn_dult_integration.h"
+#include "fp_activation.h"
+
+static struct dult_motion_detector_cb dult_motion_detector_cb;
+static bool dult_motion_detector_cb_registered;
+
+int bt_fast_pair_fmdn_motion_detector_cb_register(
+	const struct bt_fast_pair_fmdn_motion_detector_cb *cb)
+{
+	if (bt_fast_pair_is_ready() || !IS_ENABLED(CONFIG_BT_FAST_PAIR_FMDN_DULT_MOTION_DETECTOR)) {
```

Comment:
This source file is not included in case the `CONFIG_BT_FAST_PAIR_FMDN_DULT_MOTION_DETECTOR` is disabled